examples/custom_search_method/asha_search_method/asha.py

```diff
@@ -171,20 +161,20 @@ def on_trial_closed(self, request_id: uuid.UUID) -> List[Operation]:
     # Next, ASHA invoked _promote_async function to decided whether to promote
     # existing trial, start a new trial or close the experiment.
     def on_trial_exited_early(
-        self, request_id: uuid.UUID, exited_reason: ExitedReason
-    ) -> List[Operation]:
+        self, request_id: uuid.UUID, exited_reason: searcher.ExitedReason
+    ) -> List[searcher.Operation]:
         self.asha_search_state.pending_trials -= 1
 
         # The "if" statement below can be completely removed.
         # There is no need to raise INVALID_HP for hyperparameters
```

Comment:
In what scenario would the search method produce unwanted hyperparameters? 

Also, is there a way to say that if it's this hyperparameter then ignore the INVALID_HP error?
Can we control this in sample_params? By only including hyperparameter in the sample_params that we want to train?

Reply:
That is a legacy statement (maybe unnecessary). Scenario: Assume a user is running a HP search with searcher implemented in go, for instance ASHA. While user can define a range of values for selecting HP value, the user cannot exclude certain values (can be based on domain knowledge). In such scenarios, user raises INVALID_HP in their model code, and that is handled by searcher in the master (for instance, ASHA starts a new trial with new HP values).

You are absolutely right that with Custom Search method, user can simply deal with this scenario in sample_params method (which is described in the lower part of this comment). I decided to leave the code to handle INVALID_HP error for two reasons:
1. show users that raising INVALID_HP for CS can be avoided,
2. but also if their code is currently raising INVALID_HP errors, they can still handle it in the Custom Search method as they want.

How about this change: 
`Instead, make sure to start trials only with hyperparameters that satisfy your criteria. For instance, in this example, sample_params() should return only valid hyperparameters.`
?